dashboard/app/controllers/pd/mail_preview_controller.rb

```diff
@@ -0,0 +1,159 @@
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  # CSinS
+  def teacher_enrollment_receipt_cs_in_s_phase_2
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_cs_in_s_phase_3_semester_1
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_3_SEMESTER_1)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_cs_in_s_phase_3_semester_2
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_3_SEMESTER_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  # ECS
+  def teacher_enrollment_receipt_ecs_phase_2
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_PHASE_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_3
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_3)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_4
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_4)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_5
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_5)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_6
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_6)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_phase_4
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_PHASE_4)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_reminder
+    workshop = workshop(Pd::Workshop::COURSE_CSP)
+    Pd::WorkshopMailer.teacher_enrollment_reminder enrollment(workshop)
+  end
+
+  # Organizer enrollment receipt
+  def organizer_enrollment_receipt
+    workshop = workshop(Pd::Workshop::COURSE_CSF)
+    Pd::WorkshopMailer.organizer_enrollment_receipt enrollment(workshop)
+  end
+
+  # Teacher cancel receipt
+  # csf
+  def teacher_cancel_receipt_csf
+    workshop = workshop(Pd::Workshop::COURSE_CSF)
+    Pd::WorkshopMailer.teacher_cancel_receipt enrollment(workshop)
+  end
+
+  # csf
+  def teacher_cancel_receipt_other
+    workshop = workshop(Pd::Workshop::COURSE_CSP)
+    Pd::WorkshopMailer.teacher_cancel_receipt enrollment(workshop)
+  end
+
+  # Organizer cancel receipt
```

Comment:
I'm confused why there is an organizer enrollment receipt for CSF but an organizer cancel receipt for CSP?